crates/bdk/src/wallet/mod.rs

```diff
@@ -703,116 +648,159 @@ impl<D> Wallet<D> {
         self.indexed_graph.index.keychains()
     }
 
-    /// Return a derived address using the external descriptor, see [`AddressIndex`] for
-    /// available address index selection strategies. If none of the keys in the descriptor are derivable
-    /// (i.e. does not end with /*) then the same address will always be returned for any [`AddressIndex`].
-    ///
-    /// A `PersistBackend<ChangeSet>::WriteError` will result if unable to persist the new address
-    /// to the `PersistBackend`.
+    /// Peek an address of the given `keychain` at `index` without revealing it.
     ///
     /// # Panics
     ///
     /// This panics when the caller requests for an address of derivation index greater than the
-    /// BIP32 max index.
-    pub fn try_get_address(
+    /// [BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki) max index.
```

Comment:
I think it will be good to note that for non-wildcard descriptors, we will return the address at every provided index.